RestApi/Sepes.RestApi/Services/StudyService.cs

```diff
@@ -64,6 +66,42 @@ public async Task<Study> Save(Study newStudy, Study based)
             return study;
         }
 
+        public async Task<Study> DeletePod(Pod pod)
+        {
+            Study study = null;
+
+            foreach ( var iStudy in _studies.ToList())
+            {
+                if (iStudy.studyId == pod.studyId)
+                {
+                    study = iStudy;
+                }
+
+            }
+            
+            if (study.pods.Contains(pod))
+            {
+                await _podService.Delete(pod);
+                //Remove pod from study
+                var tempPods = study.pods.Remove(pod);
+                var studynew = study.ReplacePods(tempPods);
+                //Delete from database
+                await _db.UpdateStudy(studynew);
+                //Replace study in memory
+                _studies.Remove(study);
+                _studies.Add(studynew);
+                return studynew;
+                
+            }
+            else{
+                return null; //Error data mismatch, pod changed or not found. Refresh and try again
```

Comment:
This should maybe throw some custom exception. Completing a task successfully with null send mixed messages.